bftengine/src/bftengine/ReadOnlyReplica.cpp

```diff
@@ -102,58 +102,32 @@ void ReadOnlyReplica::onMessage<CheckpointMsg>(CheckpointMsg *msg) {
   ro_metrics_.received_checkpoint_msg_++;
   const ReplicaId msgGenReplicaId = msg->idOfGeneratedReplica();
   const SeqNum msgSeqNum = msg->seqNumber();
-  const Digest msgDigest = msg->digestOfState();
-  const bool msgIsStable = msg->isStableState();
   const EpochNum msgEpochNum = msg->epochNumber();
   EpochNum replicasLastKnownEpochVal = 0;
-  LOG_INFO(GL,
-           KVLOG(msg->senderId(), msgGenReplicaId, msgSeqNum, msg->size(), msgIsStable)
-               << ", digest: " << msgDigest.toString());
+  LOG_INFO(
+      GL,
+      KVLOG(msg->senderId(), msgGenReplicaId, msgSeqNum, msgEpochNum, msg->size(), msg->isStableState(), msg->state())
+          << ", state digest: " << msg->digestOfState().toString());
   // Reconfiguration cmd block is synced to RO replica via reserved pages
   auto epochNumberFromResPages = ReconfigurationCmd::instance().getReconfigurationCommandEpochNumber();
   if (epochNumberFromResPages.has_value()) {
     replicasLastKnownEpochVal = epochNumberFromResPages.value();
   }
   // not relevant
-  if (!msgIsStable || msgSeqNum <= lastExecutedSeqNum) return;
+  if (!msg->isStableState() || msgSeqNum <= lastExecutedSeqNum || msgEpochNum != replicasLastKnownEpochVal) return;
```

Comment:
Epoch number is not relevant for read-only replica

Reply:
But still there was the epoch related code - look on the left side